src/legacy/core_plugins/kibana/public/discover/directives/histogram.tsx

```diff
@@ -0,0 +1,258 @@
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiIcon, EuiSpacer } from '@elastic/eui';
+import moment from 'moment-timezone';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import lightEuiTheme from '@elastic/eui/dist/eui_theme_light.json';
+import darkEuiTheme from '@elastic/eui/dist/eui_theme_dark.json';
+
+import {
+  AnnotationDomainTypes,
+  Axis,
+  Chart,
+  HistogramBarSeries,
+  GeometryValue,
+  getAnnotationId,
+  getAxisId,
+  getSpecId,
+  LineAnnotation,
+  Position,
+  ScaleType,
+  Settings,
+  RectAnnotation,
+  TooltipValue,
+  TooltipType,
+} from '@elastic/charts';
+
+import { i18n } from '@kbn/i18n';
+
+import { getChartTheme } from 'ui/elastic_charts';
+import chrome from 'ui/chrome';
+// @ts-ignore: path dynamic for kibana
+import { timezoneProvider } from 'ui/vis/lib/timezone';
+
+export interface DiscoverHistogramProps {
+  chartData: any;
+  timefilterUpdateHandler: (ranges: { from: number; to: number }) => void;
+}
+
+export class DiscoverHistogram extends Component<DiscoverHistogramProps> {
+  public static propTypes = {
+    chartData: PropTypes.object,
+    timefilterUpdateHandler: PropTypes.func,
+  };
+
+  public onBrushEnd = (min: number, max: number) => {
+    const range = {
+      from: min,
+      to: max,
+    };
+
+    this.props.timefilterUpdateHandler(range);
+  };
+
+  public onElementClick = (xInterval: number) => (elementData: GeometryValue[]) => {
+    const startRange = elementData[0].x;
+
+    const range = {
+      from: startRange,
+      to: startRange + xInterval,
+    };
+
+    this.props.timefilterUpdateHandler(range);
+  };
+
+  public formatXValue = (val: string) => {
+    const xAxisFormat = this.props.chartData.xAxisFormat.params.pattern;
+
+    return moment(val).format(xAxisFormat);
+  };
+
+  public renderBarTooltip = (xInterval: number, domainStart: number, domainEnd: number) => (
+    headerData: TooltipValue
+  ): JSX.Element | string => {
+    const headerDataValue = headerData.value;
+    const formattedValue = this.formatXValue(headerDataValue);
+
+    const partialDataText = i18n.translate('kbn.discover.histogram.partialData.bucketTooltipText', {
+      defaultMessage:
+        'Part of this bucket may contain partial data. The selected time range does not fully cover it.',
```

Comment:
"Part of...partial data" feels a little redundant. 
```suggestion
        'The selected time range does not fully cover this bucket, it may contain partial data.',
```

Reply:
See https://github.com/elastic/kibana/pull/43788/commits/a39695966c5e6a197b421a71b31f91534f96b59e